Index: /home/toddw/as/komodo-devel/mozilla/build/moz700-ko7.9/mozilla/toolkit/locales/en-US/crashreporter/crashreporter.ini
--- a/toolkit/locales/en-US/crashreporter/crashreporter.ini	Mon Oct 24 15:33:38 2011 -0700
+++ b/toolkit/locales/en-US/crashreporter/crashreporter.ini	Wed Oct 26 10:36:32 2011 -0700
@@ -15,7 +15,7 @@
 CrashReporterDefault=This application is run after a crash to report the problem to the application vendor.  It should not be run directly.
 Details=Details…
 ViewReportTitle=Report Contents
-CommentGrayText=Add a comment (comments are publicly visible)
+CommentGrayText=Add any relevant comments
 ExtraReportInfo=This report also contains technical information about the state of the application when it crashed.
 # LOCALIZATION NOTE (CheckSendReport): The %s is replaced with the vendor name.
 CheckSendReport=Tell %s about this crash so they can fix it
Index: /home/toddw/as/komodo-devel/mozilla/build/moz700-ko7.9/mozilla/toolkit/crashreporter/content/crashes.js
--- a/toolkit/crashreporter/content/crashes.js	Mon Oct 24 15:33:38 2011 -0700
+++ b/toolkit/crashreporter/content/crashes.js	Wed Oct 26 10:58:53 2011 -0700
@@ -192,6 +192,10 @@
     else {
       link.setAttribute("href", reportURL + reports[i].id);
     }
+
+    // ACTIVESTATE - don't show submissions as links - they are not public.
+    link.removeAttribute("href");
+
     link.setAttribute("id", reports[i].id);
     link.appendChild(document.createTextNode(reports[i].id));
     cell.appendChild(link);
