Index: /home/toddw/as/komodo-devel_moz7/mozilla/build/moz700-ko7.16/mozilla/xpcom/stub/Makefile.in
--- a/xpcom/stub/Makefile.in	Wed Aug 31 10:11:17 2011 -0400
+++ b/xpcom/stub/Makefile.in	Wed Sep 28 11:47:14 2011 -0700
@@ -110,12 +110,23 @@
 DEPENDENT_LIBS_LIST += XUL
 else
 EXTRA_DSO_LIBS = xul
 DEPENDENT_LIBS_LIST += $(DLL_PREFIX)xul$(DLL_SUFFIX)
 endif
 
+# Python and PyXPCOM dependant libraries.
+ifeq ($(OS_TARGET),WINNT)
+DEPENDENT_LIBS_LIST += $(DLL_PREFIX)python26$(DLL_SUFFIX)
+else
+DEPENDENT_LIBS_LIST += $(DLL_PREFIX)python2.6$(DLL_SUFFIX)
+endif
+DEPENDENT_LIBS_LIST += \
+	$(DLL_PREFIX)xpcom$(DLL_SUFFIX) \
+	$(DLL_PREFIX)pyxpcom$(DLL_SUFFIX) \
+	$(NULL)
+
 EXTRA_DSO_LDOPTS +=				\
 	$(EXTRA_DSO_LIBS)			\
 	$(NSPR_LIBS)				\
 	$(MOZALLOC_LIB)				\
 	$(NULL)
 
