# HG changeset patch
# Parent 721c2091c42348481c524afc8b93f6a75272f1e0
# User marky
# Date 1298496565 28800

bug 37788 - force komodo to front app when started



diff --git a/toolkit/xre/nsAppRunner.cpp b/toolkit/xre/nsAppRunner.cpp
--- a/toolkit/xre/nsAppRunner.cpp
+++ b/toolkit/xre/nsAppRunner.cpp
@@ -3290,6 +3290,26 @@ XRE_main(int argc, char* argv[], const n
     }
 #endif /* MOZ_WIDGET_GTK2 */
 
+#ifdef XP_MACOSX
+        {
+         OSErr osError;
+         ProcessSerialNumber psn;
+
+         /* Make sure the window comes up in the front of the calling
+          * application.
+          */
+         if (GetCurrentProcess(&psn) >= 0) {
+             //printf("setting as front process\n");
+             osError = SetFrontProcess(&psn);
+             if (osError != noErr) {
+          printf("error setting as front process: %d!\n", osError);
+             }
+         } else {
+             printf("error: couldn't get process number\n");
+         }
+        }
+#endif
+
 #ifdef MOZ_ENABLE_XREMOTE
     // handle -remote now that xpcom is fired up
 
