Index: /as/komodo-devel/mozilla/prebuilt/python2.7/t/lib/python2.7/subprocess.py
--- lib/python2.7/subprocess.py 
+++ lib/python2.7/subprocess.py (unsaved) 
@@ -1277,9 +1277,15 @@
                     pid, sts = _waitpid(self.pid, _WNOHANG)
                     if pid == self.pid:
                         self._handle_exitstatus(sts)
-                except _os_error:
+                except _os_error as e:
                     if _deadstate is not None:
                         self.returncode = _deadstate
+                    elif e.errno == errno.ECHILD:
+                        # This happens if SIGCLD is set to be ignored or waiting
+                        # for child processes has otherwise been disabled for
+                        # our process. This child is dead, we can't get the
+                        # status.
+                        self.returncode = 0
             return self.returncode
 
 
