Index: content/canvas/src/dummy.cpp
===================================================================
RCS file: /cvsroot/mozilla/content/canvas/src/dummy.cpp,v
retrieving revision 1.1
diff -u -u -r1.1 dummy.cpp
--- content/canvas/src/dummy.cpp	18 Apr 2005 07:04:49 -0000	1.1
+++ content/canvas/src/dummy.cpp	14 Oct 2005 18:26:14 -0000
@@ -1,4 +1,8 @@
 /*
  * This file only exists because some platforms can't handle building an empty .a
  * file, or .lib, or whatever (e.g. OS X, possibly Win32).
+ *
+ * Need at least one external symbol for some linkers to create a proper
+ * archive file: https://bugzilla.mozilla.org/show_bug.cgi?id=311143
  */
+void concvsdummy(void) {}
