Index: widget/src/mac/nsMacEventHandler.cpp
===================================================================
RCS file: /cvsroot/mozilla/widget/src/mac/nsMacEventHandler.cpp,v
retrieving revision 1.172
diff -u -r1.172 nsMacEventHandler.cpp
--- widget/src/mac/nsMacEventHandler.cpp	1 Jun 2005 16:33:48 -0000	1.172
+++ widget/src/mac/nsMacEventHandler.cpp	30 Jun 2005 20:43:05 -0000
@@ -1533,6 +1533,12 @@
 			break;
 		}
 
+#if TARGET_CARBON
+    // XXX ActiveState  HIView embeded plugins that do not handle mouse events produce
+    //                  inStructure partCode's.  Handle it just like inContent, and 
+    //                  mousedown events work properly.
+    case inStructure:
+#endif
 		case inContent:
 		{
 			// don't allow clicks that rolled up a popup through to the content area.
@@ -1622,6 +1628,8 @@
 
 	nsMouseEvent mouseEvent(PR_TRUE, 0, nsnull, nsMouseEvent::eReal);
 	PRUint32 mouseButton = NS_MOUSE_LEFT_BUTTON_UP;
+	if ( aOSEvent.modifiers & controlKey )
+		  mouseButton = NS_MOUSE_RIGHT_BUTTON_UP;
 
 	// We've hacked our events to include the button.
 	// Normally message is undefined in mouse click/drag events.
Index: widget/src/mac/nsMacMessagePump.cpp
===================================================================
RCS file: /cvsroot/mozilla/widget/src/mac/nsMacMessagePump.cpp,v
retrieving revision 1.148
diff -u -r1.148 nsMacMessagePump.cpp
--- widget/src/mac/nsMacMessagePump.cpp	4 Feb 2005 22:15:08 -0000	1.148
+++ widget/src/mac/nsMacMessagePump.cpp	30 Jun 2005 20:43:05 -0000
@@ -553,6 +553,12 @@
         break;
       }
 
+#if TARGET_CARBON
+    // XXX ActiveState  HIView embeded plugins that do not handle mouse events produce
+    //                  inStructure partCode's.  Handle it just like inContent, and 
+    //                  mousedown events work properly.
+      case inStructure:
+#endif
       case inContent:
       {
         nsGraphicsUtils::SafeSetPortWindowPort(whichWindow);
