Index: nsAppFileLocationProvider.cpp
===================================================================
RCS file: /cvsroot/mozilla/xpcom/io/nsAppFileLocationProvider.cpp,v
retrieving revision 1.56
diff -u -r1.56 nsAppFileLocationProvider.cpp
--- xpcom/io/nsAppFileLocationProvider.cpp	29 Jun 2005 15:55:57 -0000	1.56
+++ xpcom/io/nsAppFileLocationProvider.cpp	25 Sep 2005 00:25:03 -0000
@@ -215,6 +223,7 @@
         if (NS_SUCCEEDED(rv))
             rv = localFile->AppendRelativeNativePath(PLUGINS_DIR_NAME);
     }
+#if 0 // ACTIVESTATE  prevent all but app dir plugins
 #if defined(XP_MAC) || defined(XP_MACOSX)
     else if (nsCRT::strcmp(prop, NS_MACOSX_USER_PLUGIN_DIR) == 0)
     {
@@ -265,6 +274,7 @@
             rv = localFile->AppendRelativeNativePath(PLUGINS_DIR_NAME);
     }
 #endif
+#endif
     else if (nsCRT::strcmp(prop, NS_APP_SEARCH_DIR) == 0)
     {
         rv = CloneMozBinDirectory(getter_AddRefs(localFile));
