Index: extensions/python/dom/src/Makefile.in
===================================================================
RCS file: /cvsroot/mozilla/extensions/python/dom/src/Makefile.in,v
retrieving revision 1.5
diff -u -r1.5 Makefile.in
--- extensions/python/dom/src/Makefile.in	10 Jan 2008 09:09:57 -0000	1.5
+++ extensions/python/dom/src/Makefile.in	15 Mar 2008 00:10:37 -0000
@@ -21,9 +21,15 @@
 
 # We always need the Python and pyxpcom includes and libs.
 LOCAL_INCLUDES = $(MOZ_PYTHON_INCLUDES)
+
+ifeq ($(OS_ARCH),Darwin)
+EXTRA_LIBS += $(MOZ_PYTHON_LIBS) \
+              $(NULL)
+else
 EXTRA_LIBS += $(MOZ_PYTHON_LIBS) \
               $(call EXPAND_LIBNAME_PATH,pyxpcom,$(DIST)/lib) \
               $(NULL)
+endif
 
 # In the shorter term we need JS too.
 EXTRA_LIBS += $(MOZ_JS_LIBS)
@@ -41,9 +47,18 @@
 
 CXXFLAGS += -DPYTHON_SO=\"libpython$(MOZ_PYTHON_VER_DOTTED).so\"
 
+ifeq ($(OS_ARCH),Darwin)
+EXTRA_DSO_LDOPTS = -bundle $(XPCOM_LIBS) \
+                    $(MOZ_COMPONENT_NSPR_LIBS) \
+                    $(XPCOM_STANDALONE_GLUE_LDOPTS) \
+                    -lpyxpcom \
+                    $(NULL)
+  
+else
 EXTRA_DSO_LDOPTS += $(XPCOM_GLUE_LDOPTS) \
                     $(MOZ_COMPONENT_LIBS) \
                     $(NULL)
+endif
 
 include $(topsrcdir)/config/rules.mk
 
