Index: extensions/spellcheck/locales/en-US/hunspell/en-US.dic
===================================================================
RCS file: /cvsroot/mozilla/extensions/spellcheck/locales/en-US/hunspell/en-US.dic,v
retrieving revision 1.5
diff -u -r1.5 en-US.dic
--- extensions/spellcheck/locales/en-US/hunspell/en-US.dic	20 Feb 2008 09:48:53 -0000	1.5
+++ extensions/spellcheck/locales/en-US/hunspell/en-US.dic	15 Mar 2008 00:03:00 -0000
@@ -134,6 +134,7 @@
 Acropolis
 Acrux/M
 Actaeon
+ActiveState/M
 Acton
 Acts
 Acuff/M
@@ -8357,6 +8358,8 @@
 Kojak/M
 Kolyma
 Kommunizma/M
+Komodo/M
+komodo/M
 Kong/M
 Kongo/M
 Konrad/M
