Index: config/autoconf.mk.in
===================================================================
RCS file: /cvsroot/mozilla/config/autoconf.mk.in,v
retrieving revision 3.363.2.19
diff -u -r3.363.2.19 config/autoconf.mk.in
--- config/autoconf.mk.in	25 Jul 2006 20:11:12 -0000	3.363.2.19
+++ config/autoconf.mk.in	1 Aug 2006 23:47:54 -0000
@@ -558,6 +558,7 @@
 MOZ_MAPINFO	= @MOZ_MAPINFO@
 
 MOZ_PHOENIX	= @MOZ_PHOENIX@
+MOZ_KOMODO	= @MOZ_KOMODO@
 MOZ_XULRUNNER	= @MOZ_XULRUNNER@
 MOZ_XUL_APP	= @MOZ_XUL_APP@
 MOZ_THUNDERBIRD = @MOZ_THUNDERBIRD@
Index: allmakefiles.sh
===================================================================
RCS file: /cvsroot/mozilla/allmakefiles.sh,v
retrieving revision 1.677
diff -u -r1.677 allmakefiles.sh
--- allmakefiles.sh	13 Jun 2007 18:02:58 -0000	1.677
+++ allmakefiles.sh	13 Jun 2007 20:05:09 -0000
@@ -836,6 +836,12 @@
 xulrunner/installer/mac/Makefile
 "
 
+MAKEFILES_komodo="
+komodo/Makefile
+komodo/app/Makefile
+komodo/app/profile/extensions/Makefile
+"
+
 MAKEFILES_xulapp="
 toolkit/Makefile
 toolkit/library/Makefile
@@ -1349,9 +1355,13 @@
     add_makefiles "$MAKEFILES_xulapp"
 fi
 
+if test -n "$MOZ_KOMODO"; then
+    add_makefiles "$MAKEFILES_komodo"
+else
 if test -n "$MOZ_XULRUNNER"; then
     add_makefiles "$MAKEFILES_xulrunner"
 fi
+fi
 
 if test -n "$MOZ_STANDALONE_COMPOSER"; then
     add_makefiles "$MAKEFILES_standalone_composer"
