Index: widget/src/cocoa/nsChildView.mm
===================================================================
RCS file: /cvsroot/mozilla/widget/src/cocoa/nsChildView.mm,v
retrieving revision 1.362
diff -u -r1.362 nsChildView.mm
--- widget/src/cocoa/nsChildView.mm	22 Jul 2008 15:19:11 -0000	1.362
+++ widget/src/cocoa/nsChildView.mm	15 Aug 2008 00:56:52 -0000
@@ -5228,6 +5228,9 @@
 - (void)keyDown:(NSEvent*)theEvent
 {
   NS_OBJC_BEGIN_TRY_ABORT_BLOCK;
+/* XXX ACTIVESTATE: allowing TSM to be used on a plugin prevents keyPress
+                    events, thus breaking our use of SciMoz.
+                    
 
   // If a plugin has the focus, we need to use an alternate method for
   // handling NSKeyDown and NSKeyUp events (otherwise Carbon-based IME won't
@@ -5247,7 +5250,7 @@
     ::TSMRemoveDocumentProperty(mPluginTSMDoc, kFocusedChildViewTSMDocPropertyTag);
     return;
   }
-
+*/
   [self processKeyDownEvent:theEvent keyEquiv:NO];
 
   NS_OBJC_END_TRY_ABORT_BLOCK;
