Index: xpfe/components/startup/src/nsAppStartup.cpp
===================================================================
RCS file: /cvsroot/mozilla/xpfe/components/startup/src/nsAppStartup.cpp,v
retrieving revision 1.7
diff -d -u -r1.7 nsAppStartup.cpp
--- xpfe/components/startup/src/nsAppStartup.cpp	10 Mar 2005 04:38:33 -0000	1.7
+++ xpfe/components/startup/src/nsAppStartup.cpp	22 Aug 2005 23:36:30 -0000
@@ -400,7 +400,8 @@
 void
 nsAppStartup::AttemptingQuit(PRBool aAttempt)
 {
-#if defined(XP_MAC) || defined(XP_MACOSX)
+#if 0
+//#if defined(XP_MAC) || defined(XP_MACOSX)
   if (aAttempt) {
     // now even the Mac wants to quit when the last window is closed
     if (!mAttemptingQuit)
Index: toolkit/components/startup/src/nsAppStartup.cpp
===================================================================
RCS file: /cvsroot/mozilla/toolkit/components/startup/src/nsAppStartup.cpp,v
retrieving revision 1.10
diff -u -r1.10 toolkit/components/startup/src/nsAppStartup.cpp
--- toolkit/components/startup/src/nsAppStartup.cpp	4 Oct 2005 14:41:18 -0000	1.10
+++ toolkit/components/startup/src/nsAppStartup.cpp	20 Jan 2006 01:18:27 -0000
@@ -336,7 +336,8 @@
 void
 nsAppStartup::AttemptingQuit(PRBool aAttempt)
 {
-#ifdef XP_MACOSX
+#if 0
+//#ifdef XP_MACOSX
   if (aAttempt) {
     // now even the Mac wants to quit when the last window is closed
     if (!mAttemptingQuit)
