Index: widget/src/mac/nsFilePicker.cpp
===================================================================
RCS file: /cvsroot/mozilla/widget/src/mac/nsFilePicker.cpp,v
retrieving revision 1.64.12.1
diff -d -u -r1.64.12.1 nsFilePicker.cpp
--- widget/src/mac/nsFilePicker.cpp	1 Sep 2005 19:18:04 -0000	1.64.12.1
+++ widget/src/mac/nsFilePicker.cpp	1 Sep 2005 22:03:11 -0000
@@ -376,6 +376,7 @@
   dialogCreateOptions.optionFlags |= kNavNoTypePopup;
   dialogCreateOptions.optionFlags |= kNavDontAutoTranslate;
   dialogCreateOptions.optionFlags |= kNavDontAddTranslateItems;
+  dialogCreateOptions.optionFlags |= kNavAllowInvisibleFiles;
   if (inAllowMultiple)
     dialogCreateOptions.optionFlags |= kNavAllowMultipleFiles;
   dialogCreateOptions.modality = kWindowModalityAppModal;
@@ -490,6 +491,7 @@
   dialogCreateOptions.optionFlags |= kNavNoTypePopup;
   dialogCreateOptions.optionFlags |= kNavDontAutoTranslate;
   dialogCreateOptions.optionFlags |= kNavDontAddTranslateItems;
+  dialogCreateOptions.optionFlags |= kNavAllowInvisibleFiles;
   dialogCreateOptions.optionFlags ^= kNavAllowMultipleFiles;
   dialogCreateOptions.modality = kWindowModalityAppModal;
   dialogCreateOptions.parentWindow = NULL;
@@ -582,6 +584,7 @@
   
   dialogCreateOptions.optionFlags |= kNavDontAutoTranslate;
   dialogCreateOptions.optionFlags |= kNavDontAddTranslateItems;
+  dialogCreateOptions.optionFlags |= kNavAllowInvisibleFiles;
   dialogCreateOptions.optionFlags ^= kNavAllowMultipleFiles;
   dialogCreateOptions.modality = kWindowModalityAppModal;
   dialogCreateOptions.parentWindow = NULL;
