Index: toolkit/xre/nsXREDirProvider.cpp
===================================================================
RCS file: /cvsroot/mozilla/toolkit/xre/nsXREDirProvider.cpp,v
retrieving revision 1.68
diff -u -r1.68 nsXREDirProvider.cpp
--- toolkit/xre/nsXREDirProvider.cpp	7 Mar 2008 01:58:18 -0000	1.68
+++ toolkit/xre/nsXREDirProvider.cpp	15 Mar 2008 01:34:09 -0000
@@ -42,6 +42,8 @@
 #include "nsAppRunner.h"
 #include "nsXREDirProvider.h"
 
+#include "prenv.h"
+
 #include "jsapi.h"
 
 #include "nsIJSContextStack.h"
@@ -1002,6 +1004,17 @@
   nsresult rv;
   nsCOMPtr<nsILocalFile> localDir;
 
+  // ActiveState Komodo patch:
+  // Allow the _XRE_USERAPPDATADIR environment variable to override.
+  const char* envvar;
+  envvar = PR_GetEnv("_XRE_USERAPPDATADIR");
+  if (envvar && *envvar) {
+    rv = NS_NewNativeLocalFile(nsDependentCString(envvar), PR_TRUE,
+                               getter_AddRefs(localDir));
+    NS_ENSURE_SUCCESS(rv, rv);
+  }
+  else {
+
 #if defined(XP_MACOSX)
   FSRef fsRef;
   OSType folderType;
@@ -1079,6 +1092,7 @@
 #else
 #error "Don't know how to get product dir on your platform"
 #endif
+  } /* if $_XRE_USERAPPDATADIR */
 
   NS_IF_ADDREF(*aFile = localDir);
   return rv;
