Disable building of marionette when ENABLE_TESTS is disabled.
Index: /as/komodo-devel/mozilla/build/moz2400-ko8.9/mozilla/toolkit/toolkit.mozbuild
--- a/toolkit/toolkit.mozbuild	Sun Jun 23 14:35:22 2013 +0300
+++ b/toolkit/toolkit.mozbuild	Mon Jul 22 10:33:14 2013 -0700
@@ -179,7 +179,7 @@
 
 DIRS += ['/addon-sdk']
 
-if CONFIG['ENABLE_MARIONETTE'] or CONFIG['MOZ_WIDGET_TOOLKIT'] not in ('gonk', 'android'):
+if CONFIG['ENABLE_TESTS'] and (CONFIG['ENABLE_MARIONETTE'] or CONFIG['MOZ_WIDGET_TOOLKIT'] not in ('gonk', 'android')):
     DIRS += ['/testing/marionette']
 
 DIRS += [
