Bug: https://bugzilla.mozilla.org/show_bug.cgi?id=1329272
Index: /home/nathan/Projects/komodo-dev/mozilla/build/moz3500-ko11.10/mozilla/build/autoconf/icu.m4
--- a/build/autoconf/icu.m4
+++ b/build/autoconf/icu.m4 (unsaved)
@@ -67,7 +67,7 @@
         fi
     fi
 
-    version=`sed -n 's/^[[:space:]]*#[[:space:]]*define[[:space:]][[:space:]]*U_ICU_VERSION_MAJOR_NUM[[:space:]][[:space:]]*\([0-9][0-9]*\)[[:space:]]*$/\1/p' "$icudir/common/unicode/uvernum.h"`
+    version=`sed -n 's/^[[[:space:]]]*#[[:space:]]*define[[:space:]][[:space:]]*U_ICU_VERSION_MAJOR_NUM[[:space:]][[:space:]]*\([0-9][0-9]*\)[[:space:]]*$/\1/p' "$icudir/common/unicode/uvernum.h"`
     if test x"$version" = x; then
        AC_MSG_ERROR([cannot determine icu version number from uvernum.h header file $lineno])
     fi

