Index: src/LexCPP.cxx
===================================================================
--- src/LexCPP.cxx	(revision 25296)
+++ src/LexCPP.cxx	(working copy)
@@ -420,8 +420,12 @@
 					levelMinCurrent = levelNext;
 				}
 				levelNext++;
-			} else if (ch == '}') {
-				levelNext--;
+			} else if (ch == '[') {
+				levelNext++;
+			} else if (ch == '}' || ch == ']') {
+				if ((levelNext & (SC_FOLDLEVELNUMBERMASK & ~SC_FOLDLEVELBASE)) > 0) {
+					levelNext--;
+				}
 			}
 		}
 		if (!IsASpace(ch))
