Index: macosx/ScintillaMacOSX.cxx
===================================================================
--- macosx/ScintillaMacOSX.cxx	(revision 27627)
+++ macosx/ScintillaMacOSX.cxx	(working copy)
@@ -655,22 +655,26 @@
         data = CFStringCreateExternalRepresentation ( kCFAllocatorDefault, string, kCFStringEncodingUnicode, 0 );
         if (data) {
             PasteboardPutItemFlavor( theClipboard, (PasteboardItemID)1, 
                               CFSTR("com.scintilla.utf16-plain-text.rectangular"),
                               data, 0 );
             CFRelease(data);
         }
     }
     data = CFStringCreateExternalRepresentation ( kCFAllocatorDefault, string, kCFStringEncodingUnicode, 0 );
     if (data) {
+        // Fixes http://bugs.activestate.com/show_bug.cgi?id=81340:
+        // data starts with a BOM, but the flavor "public.utf16-plain-text"
+        // is to be used when the data doesn't start with a BOM,
+        // "public.utf16-external-plain-text" when it does.
         PasteboardPutItemFlavor( theClipboard, (PasteboardItemID)1, 
-                                CFSTR("public.utf16-plain-text"),
+                                CFSTR("public.utf16-external-plain-text"),
                                 data, 0 );
         CFRelease(data);
         data = NULL;
     }
     data = CFStringCreateExternalRepresentation ( kCFAllocatorDefault, string, kCFStringEncodingMacRoman, 0 );
     if (data) {
         PasteboardPutItemFlavor( theClipboard, (PasteboardItemID)1, 
                                 CFSTR("com.apple.traditional-mac-plain-text"),
                                 data, 0 );
         CFRelease(data);
