Index: contrib/scintilla/lexers/LexPython.cxx
===================================================================
--- lexers/LexPython.cxx	(revision 76116)
+++ lexers/LexPython.cxx	(working copy)
@@ -94,25 +94,25 @@
 		*nextIndex = i + 1;
 
 		if (ch == '"')
 			return SCE_P_STRING;
 		else
 			return SCE_P_CHARACTER;
 	}
 }
 
 static inline bool IsAWordChar(int ch) {
-	return (ch < 0x80) && (isalnum(ch) || ch == '.' || ch == '_');
+	return ((unsigned int) ch >= 0x80) || isalnum(ch) || ch == '.' || ch == '_';
 }
 
 static inline bool IsAWordStart(int ch) {
-	return (ch < 0x80) && (isalnum(ch) || ch == '_');
+	return ((unsigned int) ch >= 0x80) || isalnum(ch) || ch == '_';
 }
 
 static void ColourisePyDoc(unsigned int startPos, int length, int initStyle,
         WordList *keywordlists[], Accessor &styler) {
 
 	int endPos = startPos + length;
 
 	// Backtrack to previous line in case need to fix its tab whinging
 	int lineCurrent = styler.GetLine(startPos);
 	if (startPos > 0) {
