Index: cocoa/ScintillaView.mm
===================================================================
--- cocoa/ScintillaView.mm	(revision 73437)
+++ cocoa/ScintillaView.mm	(working copy)
@@ -159,6 +159,7 @@
 - (BOOL) acceptsFirstMouse: (NSEvent *) theEvent
 {
 #pragma unused(theEvent)
+  return NO; // No, we want the child view to do this...
   return YES;
 }
 
@@ -169,7 +170,10 @@
  */
 - (BOOL) acceptsFirstResponder
 {
-  return YES;
+#if defined(SCINTILLA_COCOA_DEBUG)
+  fprintf(stderr, ">> ScintillaView.mm:: InnerView:: acceptsFirstResponder\n");
+  return NO; // No, we want the child view to do this...
+#endif
 }
 
 //--------------------------------------------------------------------------------------------------
@@ -179,6 +183,9 @@
  */
 - (NSMenu*) menuForEvent: (NSEvent*) theEvent
 {
+#if defined(SCINTILLA_COCOA_DEBUG)
+  fprintf(stderr, "**** We don't want to see this: ScintillaView.mm:: InnerView:: menuForEvent\n");
+#endif
   if (![mOwner respondsToSelector: @selector(menuForEvent:)])
     return mOwner.backend->CreateContextMenu(theEvent);
   else
@@ -374,6 +381,9 @@
  */
 - (void) keyDown: (NSEvent *) theEvent
 {
+#if defined(SCINTILLA_COCOA_DEBUG)
+  fprintf(stderr, "**************** >> ScintillaView.mm::InnerView::keyDown\n");
+#endif
   if (mMarkedTextRange.length == 0)
 	mOwner.backend->KeyboardInput(theEvent);
   NSArray* events = [NSArray arrayWithObject: theEvent];
@@ -823,7 +833,7 @@
     [mVerticalScroller setTarget: self];
     [mVerticalScroller setAction: @selector(scrollerAction:)];
     [self addSubview: mVerticalScroller];
-    
+
     // Establish a connection from the back end to this container so we can handle situations
     // which require our attention.
     mBackend->RegisterNotifyCallback(nil, notification);
@@ -1583,5 +1593,36 @@
 
 //--------------------------------------------------------------------------------------------------
 
+// Try specifying these events for the scintillaView, not just
+// the InnerView
+
+/**
+ * Implement the "click through" behavior by telling the caller we accept the first mouse event too.
+ */
+- (BOOL) acceptsFirstMouse: (NSEvent *) theEvent
+{
+#pragma unused(theEvent)
+#if defined(SCINTILLA_COCOA_DEBUG)
+  fprintf(stderr, ">> ScintillaView.mm:: ScintillaView:: acceptsFirstMouse\n");
+#endif
+  return NO; // No, we want the child view to do this...
+}
+
+//--------------------------------------------------------------------------------------------------
+
+/**
+ * Make this view accepting events as first responder.
+ */
+- (BOOL) acceptsFirstResponder
+{
+#if defined(SCINTILLA_COCOA_DEBUG)
+  fprintf(stderr, ">> ScintillaView.mm:: ScintillaView:: acceptsFirstResponder\n");
+#endif
+  return NO; // No, we want the child view to do this...
+}
+
 @end
 
+
+@end
+
