Send onUpdateUI notifications for scroll events - bug 100042.
===================================================================
--- cocoa/ScintillaCocoa.mm	(revision 81525)
+++ cocoa/ScintillaCocoa.mm	(working copy)
@@ -1770,6 +1770,8 @@ void ScintillaCocoa::UpdateForScroll() {
   xOffset = ptOrigin.x;
   int newTop = Platform::Minimum(ptOrigin.y / vs.lineHeight, MaxScrollPos());
   SetTopLine(newTop);
+  // Send pending scroll notifications.
+  NotifyUpdateUI();
 }
 
 //--------------------------------------------------------------------------------------------------
