Index: cocoa/ScintillaCocoa.h
===================================================================
--- cocoa/ScintillaCocoa.h	(revision 73437)
+++ cocoa/ScintillaCocoa.h	(working copy)
@@ -22,6 +22,13 @@
 #include <vector>
 #include <map>
 
+#import <Cocoa/Cocoa.h>
+#if MAC_OS_X_VERSION_MAX_ALLOWED > MAC_OS_X_VERSION_10_5
+#import <QuartzCore/CAGradientLayer.h>
+#endif
+#import <QuartzCore/CAAnimation.h>
+#import <QuartzCore/CATransaction.h>
+
 #include "ILexer.h"
 
 #ifdef SCI_LEXER
@@ -139,6 +139,8 @@
   virtual ~ScintillaCocoa();
 
   void RegisterNotifyCallback(intptr_t windowid, SciNotifyFunc callback);
+  // lower-case 'u' is deliberate.
+  void unregisterNotifyCallback();
   sptr_t WndProc(unsigned int iMessage, uptr_t wParam, sptr_t lParam);
 
   ScintillaView* TopContainer();
@@ -202,6 +209,8 @@
   void DraggingExited(id <NSDraggingInfo> info);
   bool PerformDragOperation(id <NSDraggingInfo> info);
   void DragScroll();
+  bool inDragSession() { return inDragDrop == ddDragging; };
+  bool isTracking;
   
   // Promote some methods needed for NSResponder actions.
   virtual void SelectAll();
