Index: cocoa/ScintillaCocoa.mm
===================================================================
--- cocoa/ScintillaCocoa.mm	(revision 73437)
+++ cocoa/ScintillaCocoa.mm	(working copy)
@@ -21,8 +21,6 @@
 #import <QuartzCore/CAAnimation.h>
 #import <QuartzCore/CATransaction.h>
 
-#import <Carbon/Carbon.h> // Temporary
-
 #include "ScintillaView.h"
 #include "PlatCocoa.h"
 
@@ -383,6 +381,7 @@
   timerTarget = [[[TimerTarget alloc] init: this] retain];
   layerFindIndicator = NULL;
   Initialise();
+  isTracking = false; // Never set to true?
 }
 
 //--------------------------------------------------------------------------------------------------
@@ -712,7 +711,11 @@
       return reinterpret_cast<sptr_t>(this);
       
     case SCI_GRABFOCUS:
-	  [[ContentView() window] makeFirstResponder:ContentView()];
+      // Commenting this out makes no difference -- this code wasn't called.
+#if defined(SCINTILLA_COCOA_DEBUG)
+      fprintf(stderr, "ScintillaCocoa.mm:: WndProc: don't makeFirstResponder on SCI_GRABFOCUS\n");
+#endif
+      // [[ContentView() window] makeFirstResponder:ContentView()];
       break;
 
     case SCI_SETBUFFEREDDRAW:
@@ -880,7 +883,7 @@
   if (selectedText.rectangular)
   {
     SelectionPosition selStart = sel.RangeMain().Start();
-    PasteRectangular(selStart, selectedText.s, length);
+    PasteRectangular(selStart.Position(), selectedText.s, length);
   }
   else 
     if (pdoc->InsertString(sel.RangeMain().caret.Position(), selectedText.s, length))
@@ -1083,6 +1086,10 @@
  */
 void ScintillaCocoa::StartDrag()
 {
+#ifdef KOMODO_DISABLE_DRAG_DROP
+    inDragDrop = ddNone;
+    return;
+#endif
   if (sel.Empty())
     return;
 
@@ -1525,7 +1532,7 @@
   
   // Convert absolute coordinate into the range [0..1]. Keep in mind that the visible area
   // does *not* belong to the scroll range.
-  float relativePosition = (float) topLine / MaxScrollPos();
+  float relativePosition = (float) topLine / MaxScrollPos() || 0;
   [topContainer setVerticalScrollPosition: relativePosition];
 }
 
@@ -1672,6 +1679,12 @@
   notifyProc = callback;
 }
 
+void ScintillaCocoa::unregisterNotifyCallback()
+{
+  notifyObj = NULL;
+  notifyProc = NULL;
+}
+
 //--------------------------------------------------------------------------------------------------
 
 void ScintillaCocoa::NotifyChange()
