Index: /Users/toddw/as/komodo-devel/src/scintilla/cocoa/ScintillaView.mm
===================================================================
--- cocoa/ScintillaView.mm	(revision 74856)
+++ cocoa/ScintillaView.mm	(working copy)
@@ -851,6 +868,19 @@
 
 //--------------------------------------------------------------------------------------------------
 
+- (void)setWantsLayer:(BOOL)flag
+{
+ // Set all child views to be layer *backed* views.
+  if (mContent)
+    [mContent setWantsLayer: flag];
+  if (mHorizontalScroller)
+    [mHorizontalScroller setWantsLayer: flag];
+  if (mVerticalScroller)
+    [mVerticalScroller setWantsLayer: flag];
+}
+
+//--------------------------------------------------------------------------------------------------
+
 - (void) dealloc
 {
   [mInfoBar release];
