--- lexers/LexCoffeeScript.cxx	2015-09-10 12:33:38.902720862 -0400
+++ lexers/LexCoffeeScript.cxx	2015-10-12 14:04:11.755323782 -0400
@@ -257,6 +252,10 @@
 				}
 			} else if (isoperator(static_cast<char>(sc.ch))) {
 				sc.SetState(SCE_COFFEESCRIPT_OPERATOR);
+				// Handle '..' and '...' operators correctly.
+				if (sc.ch == '.') {
+					for (int i = 0; i < 2 && sc.chNext == '.'; i++, sc.Forward()) ;
+				}
 			}
 		}
 
