--- src/EditView.cxx	2017-10-25 11:12:35.223905271 -0400
+++ src/EditView.cxx	2017-10-25 11:12:50.619904727 -0400
@@ -1736,7 +1736,7 @@
 			(!expanded && (model.foldFlags & SC_FOLDFLAG_LINEBEFORE_CONTRACTED))) {
 			PRectangle rcFoldLine = rcLine;
 			rcFoldLine.bottom = rcFoldLine.top + 1;
-			surface->FillRectangle(rcFoldLine, vsDraw.styles[STYLE_DEFAULT].fore);
+			surface->FillRectangle(rcFoldLine, vsDraw.styles[STYLE_INDENTGUIDE].fore);
 		}
 		// Paint the line below the fold
 		if ((expanded && (model.foldFlags & SC_FOLDFLAG_LINEAFTER_EXPANDED))
@@ -1744,7 +1744,7 @@
 			(!expanded && (model.foldFlags & SC_FOLDFLAG_LINEAFTER_CONTRACTED))) {
 			PRectangle rcFoldLine = rcLine;
 			rcFoldLine.top = rcFoldLine.bottom - 1;
-			surface->FillRectangle(rcFoldLine, vsDraw.styles[STYLE_DEFAULT].fore);
+			surface->FillRectangle(rcFoldLine, vsDraw.styles[STYLE_INDENTGUIDE].fore);
 		}
 	}
 }
