--- gtk/ScintillaGTK.cxx	2007-05-25 06:24:07.000000000 -0700
+++ gtk/ScintillaGTK.cxx	2007-05-25 06:23:28.000000000 -0700
@@ -1269,7 +1269,7 @@
 int ScintillaGTK::KeyDefault(int key, int modifiers) {
 	if (!(modifiers & SCI_CTRL) && !(modifiers & SCI_ALT)) {
 		if (key < 256) {
-			NotifyKey(key, modifiers);
+			AddChar(key);
 			return 0;
 		} else {
 			// Pass up to container in case it is an accelerator
