diff -r -du lexers/LexErrorList.cxx scintilla/lexers/LexErrorList.cxx
--- lexers/LexErrorList.cxx
+++ lexers/LexErrorList.cxx
@@ -292,7 +292,14 @@
 		if (AtEOL(styler, i) || (linePos >= sizeof(lineBuffer) - 1)) {
 			// End of line (or of line buffer) met, colourise it
 			lineBuffer[linePos] = '\0';
-			ColouriseErrorListLine(lineBuffer, linePos, i, styler, valueSeparate);
+
+			// ACTIVESTATE
+			// Ensure that styles are split across lines --
+			// otherwise hotspots cover "too much" and blend
+			// consecutive errors into a single hotspot.
+			ColouriseErrorListLine(lineBuffer, linePos, i - 1, styler, valueSeparate);
+			styler.ColourTo(i, SCE_ERR_DEFAULT);
+
 			linePos = 0;
 		}
 	}
