diff -u -r1.14 src/LexGen.py
--- src/LexGen.py	30 Dec 2007 22:03:23 -0000	1.14
+++ src/LexGen.py	21 May 2008 21:00:17 -0000
@@ -119,8 +119,8 @@
         out.write(updated.encode('utf-8'))
         out.close()
         print("Changed %s " % filename)
-    #~ else:
-        #~ print "Unchanged", filename
+    else:
+        print("Unchanged", filename)
 
 def Generate(inpath, outpath, commentPrefix, eolType, *lists):
     """Generate 'outpath' from 'inpath'.
@@ -271,12 +271,12 @@
         "#define VERSION_SCINTILLA \"" + versionDotted + "\"")
     UpdateLineInFile(root + "scintilla/win32/ScintRes.rc", "#define VERSION_WORDS", 
         "#define VERSION_WORDS " + versionCommad)
-    UpdateLineInFile(root + "scintilla/qt/ScintillaEditBase/ScintillaEditBase.pro",
-        "VERSION =", 
-        "VERSION = " + versionDotted)
-    UpdateLineInFile(root + "scintilla/qt/ScintillaEdit/ScintillaEdit.pro",
-        "VERSION =", 
-        "VERSION = " + versionDotted)
+    # UpdateLineInFile(root + "scintilla/qt/ScintillaEditBase/ScintillaEditBase.pro",
+    #     "VERSION =", 
+    #     "VERSION = " + versionDotted)
+    # UpdateLineInFile(root + "scintilla/qt/ScintillaEdit/ScintillaEdit.pro",
+    #     "VERSION =", 
+    #     "VERSION = " + versionDotted)
     UpdateLineInFile(root + "scintilla/doc/ScintillaDownload.html", "       Release", 
         "       Release " + versionDotted)
     UpdateLineInFile(root + "scintilla/doc/index.html",
