Index: /Users/ericp/svn/apps/komodo/src/scintilla/macosx/ScintillaMacOSX.cxx
===================================================================
--- /Users/ericp/svn/apps/komodo/src/scintilla/macosx/ScintillaMacOSX.cxx	(revision 19760)
+++ /Users/ericp/svn/apps/komodo/src/scintilla/macosx/ScintillaMacOSX.cxx	(working copy)
@@ -234,6 +234,7 @@
 ScintillaMacOSX::~ScintillaMacOSX() {
     // If the window is closed and the timer is not removed,
     // A segment violation will occur when it attempts to fire the timer next.
+    Finalise();
     if ( mouseTrackingRef != NULL ) {
         ReleaseMouseTrackingRegion(mouseTrackingRef);
     }
