Index: macosx/ScintillaMacOSX.cxx
===================================================================
--- macosx/ScintillaMacOSX.cxx	(revision 19760)
+++ macosx/ScintillaMacOSX.cxx	(working copy)
@@ -234,11 +234,11 @@
 ScintillaMacOSX::~ScintillaMacOSX() {
     // If the window is closed and the timer is not removed,
     // A segment violation will occur when it attempts to fire the timer next.
+    Finalise();
     if ( mouseTrackingRef != NULL ) {
         ReleaseMouseTrackingRegion(mouseTrackingRef);
     }
     mouseTrackingRef = NULL;
-    SetTicking(false);
 #ifdef EXT_INPUT
     ExtInput::detach (GetViewRef());
 #endif
