diff -r b2a256ceebc5 xpcom/src/PyIInterfaceInfo.cpp
--- a/xpcom/src/PyIInterfaceInfo.cpp	Wed Nov 26 13:22:07 2014 -0800
+++ b/xpcom/src/PyIInterfaceInfo.cpp	Mon Dec 01 19:59:00 2014 -0800
@@ -243,7 +243,7 @@
 	const nsXPTConstant *pRet;
 	nsresult r;
 	Py_BEGIN_ALLOW_THREADS;
-	r = pI->GetConstant(index, &pRet);
+	r = pI->GetXptConstant(index, &pRet);
 	Py_END_ALLOW_THREADS;
 	if ( NS_FAILED(r) )
 		return PyXPCOM_BuildPyException(r);
