diff -r 66f30e40a741 js/src/shell/js.cpp
--- a/js/src/shell/js.cpp	Wed Sep 26 17:06:24 2012 -0700
+++ b/js/src/shell/js.cpp	Tue Nov 06 14:03:33 2012 -0800
@@ -4758,6 +4758,14 @@
 
     if (op->getBoolOption('c'))
         compileOnly = true;
+    if (op->getBoolOption('w')) {
+        reportWarnings = JS_TRUE;
+    } else if (op->getBoolOption('W')) {
+        reportWarnings = JS_FALSE;
+    }
+    if (op->getBoolOption('s')) {
+        JS_ToggleOptions(cx, JSOPTION_STRICT);
+    }
 
     if (op->getBoolOption('m')) {
         enableMethodJit = true;
@@ -4934,6 +4942,9 @@
         || !op.addBoolOption('m', "methodjit", "Enable the JaegerMonkey method JIT")
         || !op.addBoolOption('n', "typeinfer", "Enable type inference")
         || !op.addBoolOption('c', "compileonly", "Only compile, don't run (syntax checking mode)")
+        || !op.addBoolOption('w', "warnings", "Emit warnings")
+        || !op.addBoolOption('W', "nowarnings", "Don't emit warnings")
+        || !op.addBoolOption('s', "strict", "Check strictness")
         || !op.addBoolOption('d', "debugjit", "Enable runtime debug mode for method JIT code")
         || !op.addBoolOption('a', "always-mjit",
                              "Do not try to run in the interpreter before method jitting.")
