Index: /home/toddw/as/komodo-devel_moz7/mozilla/build/moz700-ko7.16/mozilla/layout/generic/nsObjectFrame.cpp
--- a/layout/generic/nsObjectFrame.cpp	Wed Aug 31 10:11:17 2011 -0400
+++ b/layout/generic/nsObjectFrame.cpp	Wed Aug 31 13:24:31 2011 -0700
@@ -538,14 +538,15 @@
     viewMan->SetViewVisibility(view, nsViewVisibility_kShow);
   }
 
   return (needsWidget && !canCreateWidget) ? NS_ERROR_NOT_AVAILABLE : NS_OK;
 }
 
-#define EMBED_DEF_WIDTH 240
-#define EMBED_DEF_HEIGHT 200
+// XXX ActiveState
+#define EMBED_DEF_WIDTH 20
+#define EMBED_DEF_HEIGHT 20
 
 /* virtual */ nscoord
 nsObjectFrame::GetMinWidth(nsRenderingContext *aRenderingContext)
 {
   nscoord result = 0;
 
