Index: extensions/python/xpcom/src/Makefile.in
===================================================================
RCS file: /cvsroot/mozilla/extensions/python/xpcom/src/Makefile.in,v
retrieving revision 1.13
diff -u -r1.13 Makefile.in
--- extensions/python/xpcom/src/Makefile.in	10 Jan 2008 09:09:57 -0000	1.13
+++ extensions/python/xpcom/src/Makefile.in	15 Mar 2008 00:13:46 -0000
@@ -84,9 +84,18 @@
 
 CXXFLAGS += -DPYTHON_SO=\"libpython$(MOZ_PYTHON_VER_DOTTED).so\"
 
+ifeq ($(OS_ARCH),Darwin)
+# Don't use the EXTRA_DSO_LDOPTS from rules.mk, it breaks the python module
+EXTRA_DSO_LDOPTS += $(XPCOM_LIBS) \
+                    $(MOZ_COMPONENT_NSPR_LIBS) \
+                    $(XPCOM_STANDALONE_GLUE_LDOPTS) \
+                    $(NULL)
+  
+else
 EXTRA_DSO_LDOPTS += $(XPCOM_GLUE_LDOPTS) \
                     $(MOZ_COMPONENT_LIBS) \
                     $(NULL)
+endif
 
 include $(topsrcdir)/config/rules.mk
 
Index: extensions/python/xpcom/src/loader/Makefile.in
===================================================================
RCS file: /cvsroot/mozilla/extensions/python/xpcom/src/loader/Makefile.in,v
retrieving revision 1.12
diff -u -r1.12 Makefile.in
--- extensions/python/xpcom/src/loader/Makefile.in	10 Jan 2008 09:09:57 -0000	1.12
+++ extensions/python/xpcom/src/loader/Makefile.in	15 Mar 2008 00:12:13 -0000
@@ -67,9 +67,18 @@
 
 include $(topsrcdir)/config/config.mk
 
+ifeq ($(OS_ARCH),Darwin)
+# Don't use the EXTRA_DSO_LDOPTS from rules.mk, it breaks the python module
+EXTRA_DSO_LDOPTS += $(XPCOM_LIBS) \
+                    $(MOZ_COMPONENT_NSPR_LIBS) \
+                    $(XPCOM_STANDALONE_GLUE_LDOPTS) \
+                    $(NULL)
+  
+else
 EXTRA_DSO_LDOPTS += $(XPCOM_GLUE_LDOPTS) \
                     $(MOZ_COMPONENT_LIBS) \
                     $(NULL)
+endif
 
 include $(topsrcdir)/config/rules.mk
 
Index: extensions/python/xpcom/src/module/Makefile.in
===================================================================
RCS file: /cvsroot/mozilla/extensions/python/xpcom/src/module/Makefile.in,v
retrieving revision 1.6
diff -u -r1.6 Makefile.in
--- extensions/python/xpcom/src/module/Makefile.in	10 Jan 2008 09:09:57 -0000	1.6
+++ extensions/python/xpcom/src/module/Makefile.in	15 Mar 2008 00:15:39 -0000
@@ -70,9 +70,10 @@
 
 ifeq ($(OS_ARCH),Darwin)
 # Don't use the EXTRA_DSO_LDOPTS from rules.mk, it breaks the python module
-  EXTRA_DSO_LDOPTS = -bundle $(XPCOM_GLUE_LDOPTS) \
-                     $(MOZ_COMPONENT_LIBS) \
-                     $(NULL)
+EXTRA_DSO_LDOPTS = -bundle $(XPCOM_LIBS) \
+                    $(MOZ_COMPONENT_NSPR_LIBS) \
+                    $(XPCOM_STANDALONE_GLUE_LDOPTS) \
+                    $(NULL)
   
 else
   EXTRA_DSO_LDOPTS += $(XPCOM_GLUE_LDOPTS) \
