Index: xpfe/global/resources/content/hiddenWindow.xul
===================================================================
RCS file: /cvsroot/mozilla/xpfe/global/resources/content/hiddenWindow.xul,v
retrieving revision 1.41
diff -u -r1.41 hiddenWindow.xul
--- xpfe/global/resources/content/hiddenWindow.xul	10 Mar 2005 18:44:10 -0000	1.41
+++ xpfe/global/resources/content/hiddenWindow.xul	30 Jun 2005 20:32:24 -0000
@@ -37,8 +37,6 @@
 
  ***** END LICENSE BLOCK ***** -->
 
-<?xul-overlay href="chrome://navigator/content/navigatorOverlay.xul"?>
-
 <!-- hiddenwindow is a "minimal" XUL window intended for creating the,
      er, hidden window.  This window is never shown, but on platforms
      which leave the app running after the last (visible) window is shut
@@ -68,45 +66,4 @@
   titlemenuseparator="&mainWindow.titlemodifiermenuseparator;"
   onload="hiddenWindowStartup();">
 
-  <!-- JS from navigator.xul, to handle the menu commands -->
-  <script type="application/x-javascript" src="chrome://global/content/strres.js"/>
-  <script type="application/x-javascript" src="chrome://navigator/content/browser.js"/>
-  <script type="application/x-javascript" src="chrome://navigator/content/navigator.js"/>
-
-  <!-- Shared Bookmarks Utility Library -->
-  <script type="application/x-javascript" src="chrome://communicator/content/bookmarks/bookmarks.js"/>
-  <script type="application/x-javascript" src="chrome://communicator/content/bookmarks/bookmarksMenu.js"/>
-
-  <stringbundleset id="stringbundleset">
-    <stringbundle id="bundle_navigator" src="chrome://navigator/locale/navigator.properties"/>
-  </stringbundleset>
-
-  <!-- keys are appended from the overlay -->
-  <keyset id="tasksKeys">
-    <keyset id="navKeys">
-      <key id="showHideSidebar"/>
-    </keyset>
-  </keyset>
-
-  <!-- commands are appended from the overlay -->
-  <commandset id="commands">
-    <commandset id="tasksCommands"/>
-    <commandset id="globalEditMenuItems"/>
-    <commandset id="selectEditMenuItems"/>
-    <commandset id="undoEditMenuItems"/>
-    <commandset id="clipboardEditMenuItems"/>
-  </commandset>
-
-  <broadcasterset id="navBroadcasters"/>
-
-  <template id="bookmarksMenuTemplate"/>
-
-  <!-- it's the whole navigator.xul menubar! hidden windows need to
-       have a menubar for situations where they're the only window remaining
-       on a platform that wants to leave the app running, like the Mac.
-  -->
-  <toolbox id="toolbox">
-    <menubar id="main-menubar" position="1"/>
-  </toolbox>
-
 </window>
