Index: Win32/libpcre.mak
===================================================================
--- Win32/libpcre.mak	(revision 39598)
+++ Win32/libpcre.mak	(working copy)
@@ -70,21 +70,21 @@
 	-@erase "$(INTDIR)\pcre_xclass.obj"
 	-@erase "$(INTDIR)\pcreposix.obj"
 	-@erase "$(INTDIR)\vc60.idb"
 	-@erase "$(OUTDIR)\libpcre.lib"
 
 "$(OUTDIR)" :
     if not exist "$(OUTDIR)/$(NULL)" mkdir "$(OUTDIR)"
 
 MTL=midl.exe
 CPP=cl.exe
-CPP_PROJ=/nologo /MT /W3 /GX /O2 /I "Win32" /I "." /D "WIN32" /D "NDEBUG" /D "_MBCS" /D "_LIB" /D "PCRE_STATIC" /Fp"$(INTDIR)\libpcre.pch" /YX /Fo"$(INTDIR)\\" /Fd"$(INTDIR)\\" /FD /c 
+CPP_PROJ=/nologo /MT /W3 /EHsc /O2 /I "Win32" /I "." /D "WIN32" /D "NDEBUG" /D "_MBCS" /D "_LIB" /D "PCRE_STATIC" /Fp"$(INTDIR)\libpcre.pch" /Fo"$(INTDIR)\\" /Fd"$(INTDIR)\\" /FD /c 
 
 .c{$(INTDIR)}.obj::
    $(CPP) @<<
    $(CPP_PROJ) $< 
 <<
 
 .cpp{$(INTDIR)}.obj::
    $(CPP) @<<
    $(CPP_PROJ) $< 
 <<
@@ -190,21 +190,21 @@
 	-@erase "$(INTDIR)\pcreposix.obj"
 	-@erase "$(INTDIR)\vc60.idb"
 	-@erase "$(INTDIR)\vc60.pdb"
 	-@erase "$(OUTDIR)\libpcre.lib"
 
 "$(OUTDIR)" :
     if not exist "$(OUTDIR)/$(NULL)" mkdir "$(OUTDIR)"
 
 MTL=midl.exe
 CPP=cl.exe
-CPP_PROJ=/nologo /MTd /W3 /Gm /GX /ZI /Od /I "Win32" /I "." /D "WIN32" /D "_DEBUG" /D "_MBCS" /D "_LIB" /D "PCRE_STATIC" /Fp"$(INTDIR)\libpcre.pch" /YX /Fo"$(INTDIR)\\" /Fd"$(INTDIR)\\" /FD /GZ /c 
+CPP_PROJ=/nologo /MTd /W3 /Gm /EHsc /ZI /Od /I "Win32" /I "." /D "WIN32" /D "_DEBUG" /D "_MBCS" /D "_LIB" /D "PCRE_STATIC" /Fp"$(INTDIR)\libpcre.pch" /Fo"$(INTDIR)\\" /Fd"$(INTDIR)\\" /FD /GZ /c 
 
 .c{$(INTDIR)}.obj::
    $(CPP) @<<
    $(CPP_PROJ) $< 
 <<
 
 .cpp{$(INTDIR)}.obj::
    $(CPP) @<<
    $(CPP_PROJ) $< 
 <<
